datums_warehouse/storage.py

```diff
@@ -53,6 +57,37 @@ def _write_csv(self, df, itv, prv):
         else:
             prv.unlink()
 
+    def last_time_of(self, interval):
+        _, last_time = self._get_last_of(interval, until=None)
+        return last_time
+
+    def _get_last_of(self, interval, until):
+        last_time = None
+        last_df = None
+        for df, _ in self._all_of(interval):
+            if until and df.timestamp.iloc[0] > until:
+                continue
```

Comment:
This doesn't catch the case when another data frame has a last timestamp past the util. i.e. it would select anything after until which is not the behaviour that would be expected. Probable another check for early braking is necessary and an appropriate test

Reply:
It does, as it checks if the data frame actually starts higher than the until parameters and is then ignored